starlette/convertors.py

```diff
@@ -1,67 +1,66 @@
 import math
-import typing
+from typing import Any, Dict, Generic, TypeVar
 
+T = TypeVar("T")
 
-class Convertor:
-    regex = ""
 
-    def convert(self, value: str) -> typing.Any:
+class Convertor(Generic[T]):
+    regex = r""
+
+    def convert(self, value: str) -> T:
         raise NotImplementedError()  # pragma: no cover
 
-    def to_string(self, value: typing.Any) -> str:
+    def to_string(self, value: Any) -> str:
```

Comment:
It's not really clear to me what the type of `value` is. Possibly it can also be `T`, but the semantics of `to_string` are not documented so I'm not sure.